components/AuthProvider.tsx

```diff
@@ -0,0 +1,75 @@
+import { UserDoc } from 'models/users'
+import { onAuthStateChanged, User } from 'firebase/auth'
+import { doc, onSnapshot } from 'firebase/firestore'
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from 'react'
+import { auth, userRef } from 'utils/firebase'
+
+const authContext = createContext<{
+  user: User | null
+  userLoading: boolean
+  userData: UserDoc | null
+  userDataLoading: boolean
+}>({
+  user: null,
+  userLoading: true,
+  userData: null,
+  userDataLoading: true,
+})
+
+export const useUser = () => {
+  return useContext(authContext)
+}
+
+//There is a distinction between user and userData
+//User is the data used by Firebase to distinguish different user accounts
+//userData is the data we use to handle list ownership
```

Comment:
Maybe just call `userData` `userDoc`? Since it's a user Firestore doc and I feel like that's less confusing?